default/AI/AIDependencies.py

```diff
@@ -290,3 +290,14 @@
     PART_KRILL_SPAWNER: {STEALTH_MODIFIER: 40, STACKING_RULES: [NO_EFFECT_WITH_CLOAKS]}
 }
 
+WEAPON_UPGRADE_DICT = {
+    # "PARTNAME": tuple([  (tech_name, dmg_upgrade), (tech_name2, dmg_upgrade2), ... ])
+    "SR_WEAPON_1_1": tuple([("SHP_WEAPON_1_%d" % i, 1) for i in [2, 3, 4]]),
```

Comment:
You can simplify it by replacing list comprehansion to generator expression
`"SR_WEAPON_1_1": tuple(("SHP_WEAPON_1_%d" % i, 1) for i in [2, 3, 4]),`

And replasing list to string:
`"SR_WEAPON_1_1": tuple(("SHP_WEAPON_1_%s" % i, 1) for i in '234'),`